reana_workflow_controller/rest.py

```diff
@@ -1526,3 +1528,55 @@ def _get_current_job_progress(workflow_id):
                 'prettified_cmd': job.prettified_cmd,
                 'current_job_name': job.name}
     return current_job_commands
+
+
+def _delete_workflow(workflow, all_runs=False, hard_delete=False):
+    """Delete workflow."""
+    if workflow.status in [WorkflowStatus.created,
+                           WorkflowStatus.finished,
+                           # WorkflowStatus.stopped
+                           WorkflowStatus.failed]:
+        to_be_deleted = [workflow]
+        if all_runs:
+            to_be_deleted += Session.query(Workflow).\
+                filter_by(name=workflow.name).all()
+        for workflow in to_be_deleted:
+            if hard_delete:
+                _hard_delete_workflow(workflow)
+            else:
+                _soft_delete_workflow(workflow)
+
+        return jsonify({'message': 'Workflow successfully deleted',
```

Comment:
Could we just return the dict and let the view decide what to do with it? I know there are other places where this is being done, but I would rather not do it.

Reply:
Sure, which dict do you mean? I was thinking of returning a list of dicts in the case that more than one workflow was deleted. It could be done with just a field:
```
jsonify({'deleted_workflows': [w.1, w.2, w.3]})
```
Are you thinking something different?

Reply:
I was thinking more about returning always the list of deleted workflows, and then inside the view `set_workflow_status` we would format the user message.

Reply:
Maybe you have this change locally and you didn't push it yet?

Reply:
I think we should do this change, and it will be more obvious when the utility functions move to their own utils.py, but I would not include it in this PR as it is currently in a consistent state (with the improvement pending, but still consistent). In the refactoring of the utils, we can have every non view method return the right response.